File: resources/ios/AlertDialogControl.swift
import SwiftUI

struct AlertDialogControl: View {
    @Binding var state: AlertDialogRendererState
    let onDismiss: () -> Void

    private var configuration: AlertDialogRendererConfiguration {
        state.configuration
    }

    var body: some View {
        Color.clear
            .frame(width: 0, height: 0)
            .alert(
                configuration.title,
                isPresented: Binding(
                    get: { state.isPresented },
                    set: { presented in
                        if !presented { onDismiss() }
                    }
                )
            ) {
                Button(configuration.actionLabel) {
                    onDismiss()
                }
            } message: {
                Text(configuration.message)
            }
    }
}

File: resources/android/AlertDialogControl.kt
package dev.firstlightui.plugins.firstlight_ui.ui

import androidx.compose.material3.AlertDialog
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.nativephp.plugins.native_ui.NativeUITokens

@Composable
internal fun AlertDialogControl(
    state: AlertDialogRendererState,
    tokens: NativeUITokens,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier,
) {
    if (!state.isPresented) return

    val configuration = state.configuration

    AlertDialog(
        onDismissRequest = onDismiss,
        confirmButton = {
            TextButton(
                onClick = onDismiss,
                colors = ButtonDefaults.textButtonColors(
                    contentColor = tokens.primary,
                ),
            ) {
                Text(configuration.actionLabel)
            }
        },
        title = { Text(configuration.title) },
        text = { Text(configuration.message) },
        modifier = modifier,
    )
}
